src/index.ts

```diff
@@ -1,3 +1,89 @@
-export default function helloWorld(): string {
-	return 'Hello world!';
-}
+// Imports
+import {
+	CSSValidator,
+	ValidateTextOptions,
+	ValidateTextResultBase,
+	ValidateTextResultWithWarnings,
+	W3CCSSValidatorResponse,
+} from './types';
+import * as https from 'https';
+
+// Validates CSS using W3C's public CSS validator service
+const cssValidator: CSSValidator = {
+	// Validates a string of CSS
+	async validateText(unvalidatedStr: string, options?: ValidateTextOptions): Promise<any> {
+		// Build URL for fetching
+		const url = `https://jigsaw.w3.org/css-validator/validator?text=${encodeURIComponent(unvalidatedStr)}&usermedium=${
+			options?.medium ?? 'all'
+		}&warning=${options?.warningLevel ?? 0}&output=application/json`;
+
+		// Build result and initialize response
+		const base: ValidateTextResultBase = {
+			valid: false,
+			errors: [],
+		};
+		const result: ValidateTextResultWithWarnings | ValidateTextResultBase = options?.warningLevel
+			? {
+					...base,
+					warnings: [],
+			  }
+			: base;
+		let cssValidationResponse: W3CCSSValidatorResponse['cssvalidation'] | null = null;
+
+		// Detect if fetch is available and retrieve validation response accordingly
+		if (typeof window?.fetch === 'function') {
+			const res = await fetch(url);
+			const data = (await res.json()) as W3CCSSValidatorResponse;
+
+			cssValidationResponse = data.cssvalidation;
+		} else {
+			const retrieveValidationResponse = async (): Promise<W3CCSSValidatorResponse['cssvalidation']> => {
+				return new Promise((resolve) => {
+					https.get(url, (res) => {
+						let data = '';
+
+						res.on('data', (chunk) => {
+							data += chunk;
+						});
+
+						res.on('end', () => {
+							resolve((JSON.parse(data) as W3CCSSValidatorResponse).cssvalidation);
+						});
+					});
+				});
+			};
+
+			cssValidationResponse = await retrieveValidationResponse();
+		}
```

Comment:
This should probably be its own function; then we can add more commenting and break it up a bit more. Also, that should allow us to define `cssValidationResponse` as a `const`, and avoid the `| null` type.